### PR TITLE
Storyshots: Fix mdx transform

### PR DESCRIPTION
**Pull request by rileylnapier**
after we transform from mdx to js... we need to name the file extension js so our other jest transforms transform the js correctly.

Issue:

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: addons/docs/jest-transform-mdx.js [hunk 1]
```diff
@@ -29,6 +29,10 @@ module.exports = {
       import { mdx } from '@mdx-js/react'
       ${mdx.sync(src, { compilers, filepath: filename })}
     `;
-    return getNextTransformer(filename, config).transformSource(filename, result, instrument);
+    
+    const extension = path.extname(filename);
+    const jsFileName = `${filename.slice(0, -extension.length)}.js`;
```
Comment (shilman) [inline addons/docs/jest-transform-mdx.js]:
```suggestion
    const jsFileName = `${filename}.js`;
```

Comment (shilman) [reply addons/docs/jest-transform-mdx.js]: WDYT about this?

**Review by shilman — COMMENTED** [review]
Makes sense, thank you for the fix! 🙏 Please see my comment inline.

File: addons/docs/jest-transform-mdx.js [hunk 2]
```diff
@@ -29,6 +29,9 @@ module.exports = {
       import { mdx } from '@mdx-js/react'
       ${mdx.sync(src, { compilers, filepath: filename })}
     `;
-    return getNextTransformer(filename, config).transformSource(filename, result, instrument);
+    
+    const jsFileName = `${filename}.js`;
```
Comment (rileylnapier) [inline addons/docs/jest-transform-mdx.js]: wait this might not work

Comment (rileylnapier) [reply addons/docs/jest-transform-mdx.js]: nvm, it does. the file will be index.stories.mdx.js though which i think is fine

**Comment by shilman** [comment]
@rileylnapier Hmm this seems to break CI. @Hypnosphi can you take a quick look?

File: addons/docs/jest-transform-mdx.js (hunk 1 above)
Comment (Hypnosphi) [reply addons/docs/jest-transform-mdx.js]: Shouldn't we also update https://github.com/storybookjs/storybook/blob/c6d4f69d2895e063acdf6c4abdeb762fa1142a26/addons/docs/jest-transform-mdx.js#L12 then? Not sure if it'll help though

**Comment by Hypnosphi** [comment]
@rileylnapier can you please describe the original issue you're solving here? Right now the JS transforms are applied to mdx, it's done in `getNextTransformer`

File: addons/docs/jest-transform-mdx.js (hunk 1 above)
Comment (Hypnosphi) [reply addons/docs/jest-transform-mdx.js]: Looks like this change is what actually breaks the tests. `foo.stories.mdx.js` doesn't match anything from our `transform` patterns in `jest.config.js`

**Comment by rileylnapier** [comment]
> @rileylnapier can you please describe the original issue you're solving here? Right now the JS transforms are applied to mdx, it's done in `getNextTransformer`

The issue that I had is that after my mdx gets transformed into JS.  The JS that results has things like `import React from "react"` and then i get an error because I still need to transpile this JS... but my jest config wont' transpile it because it still has a .mdx extention.

**Comment by Hypnosphi** [comment]
@rileylnapier can you please share your jest config? Or, even better, create a reproduction repo? We have JS transforms in our [config](https://github.com/storybookjs/storybook/blob/next/jest.config.js#L44-L45), and they're definitely applied

**Comment by rileylnapier** [comment]
this was my config:

```
module.exports = {
  preset: "ts-jest",
  testEnvironment: "jsdom",
  roots: ["<rootDir>"],
  transform: {
    "\\.mdx$": "@storybook/addon-docs/jest-transform-mdx",
    "\\.(ts|tsx|js|jsx)$": "ts-jest",
  },
  transformIgnorePatterns: ["node_modules"],
  moduleFileExtensions: ["js", "json", "jsx", "ts", "tsx", "mdx"],
  moduleNameMapper: {
    "^[@./a-zA-Z0-9$_-]+\\.(png|jpg|gif)$": "RelativeImageStub",
  },
  setupFilesAfterEnv: ["<rootDir>/../../setup/setupTest.js"],
  setupFiles: ["<rootDir>/../../setup/jest.init.js"],
  globals: {
    "ts-jest": {
      tsConfig: "<rootDir>/.storybook/tsconfig.json",
    },
  },
};
```

**Comment by Hypnosphi** [comment]
@rileylnapier can you please also copy a few lines above your screenshots? Looks like they're saying something about ts-jest

File: addons/docs/jest-transform-mdx.js (hunk 2 above)
Comment (Hypnosphi) [inline addons/docs/jest-transform-mdx.js]:
```suggestion
    const extension = path.extname(filename);
    const jsFileName = `${filename.slice(0, -extension.length)}.js`;
```

Needed to fix tests, see my comment in conversation above

**Comment by rileylnapier** [comment]
ts-jest[jest-transformer] (WARN) Got a unknown file type to compile (file: /Users/rileynapier/trycourier/frontend/packages/studio
/components/atoms/form/checkbox/index.stories.mdx). To fix this, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match this kind of files anymore.


I guess this is why i needed the file to be .js so tsjest would transpile it after?

File: addons/docs/jest-transform-mdx.js (hunk 1 above)
Comment (rileylnapier) [reply addons/docs/jest-transform-mdx.js]: ahhh so i had it right the first time?  :)

**Comment by Hypnosphi** [comment]
OK I got it, your solution works better indeed. Actually, we can replace `getNextTransformer` with just `new ScriptTransformer(config)` then, as we don't need to patch `transform` array anymore

File: addons/docs/jest-transform-mdx.js [hunk 3]
```diff
@@ -29,6 +29,9 @@ module.exports = {
...
+    
+    return getNextTransformer(filename, config).transformSource(jsFileName, result, instrument);
```
Comment (Hypnosphi) [inline addons/docs/jest-transform-mdx.js]:
```suggestion
    return new ScriptTransformer(config).transformSource(jsFileName, result, instrument);
```

Comment (rileylnapier) [reply addons/docs/jest-transform-mdx.js]: we would want to delete `getNextTransformer` then right?

Comment (Hypnosphi) [reply addons/docs/jest-transform-mdx.js]: yep

**Review by Hypnosphi — APPROVED** [review]

**Comment by rileylnapier** [comment]
so to be clear.   this was not an issue with babel right?  because babel doesn't have file extension restrictions right?  so you tell babel to transpile a file with .mdx and say its js, babel is cool with that.

however, something like ts-jest does care about the file extension and giving it a .mdx file it gets mad.

i did have to enable `allowJs` inside the tsConfig for this too btw

**Pull request merged**